From 5df9a5e36b7baa969c577ad23bab8d23e53c1e09 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 15:13:20 +0530
Subject: [PATCH 034/500] OpenWrt:
 465-m25p80-mx-disable-software-protection.patch

Disable software protection bits for Macronix flashes.

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: I127e2dc125acbdaa64f7e17e2c47dec65a94f888
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 2b468b530cf369c068d13da01aba4bd35290d5d2)
---
 drivers/mtd/spi-nor/macronix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/spi-nor/macronix.c b/drivers/mtd/spi-nor/macronix.c
index eb149e517c1f..283c679932f2 100644
--- a/drivers/mtd/spi-nor/macronix.c
+++ b/drivers/mtd/spi-nor/macronix.c
@@ -108,6 +108,7 @@ static const struct flash_info macronix_nor_parts[] = {
 static void macronix_nor_default_init(struct spi_nor *nor)
 {
 	nor->params->quad_enable = spi_nor_sr1_bit6_quad_enable;
+	nor->flags |= SNOR_F_HAS_LOCK;
 }
 
 static int macronix_nor_late_init(struct spi_nor *nor)
-- 
2.34.1

